src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -426,6 +427,22 @@ export function getInputType(
 
             models.set(name, model);
 
+            if(discriminator) {
+                const discriminatorProp = {
+                    Name: discriminator.propertyName,
+                    SerializedName: discriminator.propertyName,
+                    Description: "",
+                    Type: {
+                        Name: "String",
+                        Kind: InputTypeKind.String,
+                        IsNullable: false
+                    } as InputPrimitiveType,
+                    IsRequired: true,
+                    IsReadOnly: false,
+                    IsDiscriminator: true
+                };
+                properties.push(discriminatorProp);
```

Comment:
If the discriminator is related to the convinience API, better to verify all derived models are also generated.

Reply:
I will collect usages for derived models in other PR.